### PR TITLE
Fix wasm demo

### PR DESCRIPTION
**Pull request by arv**
The setup code was required to run before main. Refactor to not share
globals and let the idb js have no side effects.

### CODE REVIEW

File: idb/idb_js.go [hunk 1]
```diff
@@ -15,20 +15,24 @@ const (
...
 
-replicache.openIDB = async function(dbname) {
-    return await idb.openDB(dbname, 1, {
-      upgrade(db) {
-          db.createObjectStore("chunks");
-        },
-    });
-};`)
+func setup() {
+	idb = eval(idbCode)
```
Comment (arv) [inline idb/idb_js.go:27]: eval returns the completion value. This allows us to evaluate the code without adding global variables.

File: idb/idb_js.go [hunk 2]
```diff
@@ -15,20 +15,24 @@ const (
...
+	openIDB = eval(`
```
Comment (arv) [inline idb/idb_js.go:28]: openIDB was only used in this Go file so I used a non global function

**Comment by arv** [comment]
@whiten The main reason for this was that repjs/js.go throws on startup because the idb setup was not called yet.

I also wanted to play with this as you might imagine...

File: idb/idb_js.go (hunk 2 above)
Comment (whiten) [reply idb/idb_js.go:28]: 👍

**Review by whiten — APPROVED** [review]

**Pull request merged**